### ISSUE TITLE
Exception: No softmax found.

### ISSUE THREAD

**Issue by mahdidrm** [uchidalab/softmaxgradient-lrp#3]
Hello. thank you very much for your perfect algorithm. I have an error that confused me a lot:
This is my model:
```
input_shape=(256,256,3)

model = Sequential()
model.add(Conv2D(6, (5, 5), input_shape=input_shape, padding='same', activation = 'relu'))
model.add(MaxPooling2D(pool_size=(2, 2)))

model.add(Conv2D(16, (5, 5), padding='same', activation = 'relu'))
model.add(Activation('relu'))
model.add(MaxPooling2D(pool_size=(2, 2)))

model.add(Conv2D(64, (3, 3), activation = 'relu'))
model.add(MaxPooling2D(pool_size=(2, 2)))

model.add(Flatten())
model.add(Dense(128, activation = 'relu'))
model.add(Dropout(0.5))
model.add(Dense(7, activation = 'softmax'))

model.compile(loss='categorical_crossentropy', metrics=['accuracy'],optimizer='adam')
```
- But when I want to use of LRP I got the error: 
```
import innvestigate
import innvestigate.utils

# Get model
model1 =model
# Strip softmax layer
model = innvestigate.utils.model_wo_softmax(model1)
```
- Output:
Exception: No softmax found.

- I am using of keras 2.2.4 and tensorflow 1.17.0 

Could you please help me to solve it?

**Comment by Yung-zi** [comment]
I run the code in google colab. However, an error occurred shown as "ERROR: Could not find a version that satisfies the requirement tensorflow==1.10.0 (from versions: none)". I tried so many methods but they were failed. I saw the code environment is  python3.5, tensorflow 1.10, these versions are a little old. Why don't use an upgrading version?

**Comment by mahdidrm** [comment]
> I run the code in google colab. However, an error occurred shown as "ERROR: Could not find a version that satisfies the requirement tensorflow==1.10.0 (from versions: none)". I tried so many methods but they were failed. I saw the code environment is python3.5, tensorflow 1.10, these versions are a little old. Why don't use an upgrading version?

They will never answer! 

I think the version of tensorflow in colab is 2. It mabe helpful if you uninstall and reinstall the needed tensorflow.

**Comment by biwana** [comment]
The code from this repository uses essential functions from https://github.com/albermax/innvestigate
iNNvestigate is also based on Tensorflow 1.12. So, this repository might work on TF 1.12 code or lower. 

To Mahdidrm, I have had no problem with Dense(nb_class, activation='softmax'). You can try your model in the docker environment in /docker. It should work.
